packages/react/src/components/Slider/Slider.tsx

```diff
@@ -0,0 +1,276 @@
+import * as React from 'react'
+import * as _ from 'lodash'
+import * as PropTypes from 'prop-types'
+import * as customPropTypes from '@stardust-ui/react-proptypes'
+import { handleRef, Ref } from '@stardust-ui/react-component-ref'
+import cx from 'classnames'
+
+import {
+  applyAccessibilityKeyHandlers,
+  AutoControlledComponent,
+  ChildrenComponentProps,
+  commonPropTypes,
+  isFromKeyboard,
+  partitionHTMLProps,
+  UIComponentProps,
+  RenderResultConfig,
+  setWhatInputSource,
+} from '../../lib'
+import {
+  ComponentEventHandler,
+  ShorthandValue,
+  WithAsProp,
+  withSafeTypeForAs,
+  Omit,
+} from '../../types'
+import { Accessibility } from '../../lib/accessibility/types'
+import { sliderBehavior } from '../../lib/accessibility'
+import { SupportedIntrinsicInputProps } from '../../lib/htmlPropsUtils'
+import Box from '../Box/Box'
+
+const defaultPropValues = {
+  max: 100,
+  min: 0,
+  step: 1,
+  value: 50,
+}
+
+const processInputValues = (
+  p: Pick<SliderProps, 'min' | 'max'> & Pick<SliderState, 'value'>,
+): { min: number; max: number; value: number; valueAsPercentage: string } => {
+  let min = _.toNumber(p.min)
+  let max = _.toNumber(p.max)
+  let value = _.toNumber(p.value)
+
+  if (isNaN(min)) min = defaultPropValues.min
+  if (isNaN(max)) max = defaultPropValues.max
+  value = isNaN(value) ? defaultPropValues.value : Math.min(max, Math.max(min, value))
+  const valueAsPercentage = `${(100 * (value - min)) / (max - min)}%`
+
+  return { min, max, value, valueAsPercentage }
+}
+
+export interface SliderSlotClassNames {
+  input: string
+  inputWrapper: string
+  rail: string
+  thumb: string
+  track: string
+}
+
+export interface SliderProps
+  extends UIComponentProps,
+    ChildrenComponentProps,
+    Omit<SupportedIntrinsicInputProps, 'defaultValue'> {
```

Comment:
I understand that we want to reuse these handlers, but it seems a bit awkward why on some handlers, like `onChange` we provide the props, but on others, like `onMouseDown` we don't... Can we somehow unify this behavior?

Reply:
no idea how, but I don't think it's a blocker for this PR
I added the `onChange` here just because it's very important for the `Slider` (same is done in `Input`)...